gammapy/spectrum/models.py

```diff
@@ -2,24 +2,90 @@
 """Spectral models for Gammapy.
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
+from ..extern.bunch import Bunch
+from . import CountsSpectrum
+import numpy as np
+
 
 __all__ = [
     'SpectralModel',
     'PowerLaw',
     'ExponentialCutoffPowerLaw',
 ]
 
+# Note: Consider to move stuff from _models_old.py here
 
 class SpectralModel(object):
     """Spectral model base class.
     """
+    def __call__(self, energy):
+        kwargs = self.parameters
+        kwargs.update(energy=energy)
+        return self.evaluate(**kwargs)
+
+    def predicted_counts(self, livetime, aeff, edisp):
```

Comment:
I don't think the spectral model class is a good location for `predicted_counts`.

How about putting it here, taking a spectral model as input (i.e. not needing an import)?
http://docs.gammapy.org/en/latest/api/gammapy.data.DataStoreObservation.html

The advantage is that this `Observation` class has access to IRFs and livetime, and the livetime / GTI handling will get a little more complicated, so it's better to have the `npred` code there I think.